src/main/kotlin/gg/essential/vigilance/gui/SettingsGui.kt

```diff
@@ -184,7 +184,16 @@ class SettingsGui(
         newCategory.scrollToTop()
         currentCategory = newCategory
 
-        sidebarScroller.childrenOfType<CategoryLabel>().firstOrNull { it.isSelected }?.deselect()
+        val labels = sidebarScroller.childrenOfType<CategoryLabel>()
+
+        labels
+            .firstOrNull { it.isSelected }
+            ?.deselect()
+
+        // We compare names because the `items` field could technically be the same but the `Category` could be a different instance.
```

Comment:
But can the same argument not be made for `name` as well? Shouldn't we just compare by reference?